src/utils/inferencedata.jl

```diff
@@ -1,17 +1,83 @@
-using InferenceObjects
-using PosteriorDB
+function inferencedata(m::SampleModel; kwargs...)
 
-import Base: convert
+    # Used keyword arguments
+    used_keywords = [:include_log_likelihood, :include_sample_stats,
+        :include_posterior_predictive]
 
-"""
+    # Default is only posterior values
+    include_log_likelihood = false
+    include_sample_stats = false
+    include_posterior_predictive = false
 
-# inference
+    # Warmup included if stored
+    include_warmup = m.save_warmup
 
-# Convert the output file(s) created by cmdstan to a InferenceData object.
+    for (indx, kwd) in enumerate(kwargs)
+        if kwd[1] == :include_log_likelihood
+            include_log_likelihood = kwargs[indx][1] ? true : false
+        end
+        if kwd[1] == :include_sample_stats
+            include_sample_stats = kwargs[indx][1] ? true : false
+        end
+        if kwd[1] == :include_posterior_predictive
+            include_posterior_predictive = kwargs[indx][1] ? true : false
+        end
+    end
 
-$(SIGNATURES)
+    stan_nts = read_samples(m, :namedtuples; include_internals=include_sample_stats)
+    idata = from_namedtuple(stan_nts[[:mu, :theta_tilde, :theta, :tau]])
```

Comment:
Adding `kwargs` here allows the user to have some control over how `from_namedtuple` does conversion, e.g. to provide `dims` and `coords`.
```suggestion
    idata = from_namedtuple(stan_nts; kwargs...)
```